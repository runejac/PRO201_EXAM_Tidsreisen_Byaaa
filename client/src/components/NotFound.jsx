import React from "react";
import { Modal } from "react-bootstrap";
import { CapsuleButtonYellow } from "./CapsuleButton.jsx";
import { useNavigate } from "react-router-dom";
import "../css/notFound.css";

export function NotFound() {
  const navigate = useNavigate();
  return (
    <div id={"background-modal-notfound"}>
      <Modal
        className={"d-flex justify-content-center align-items-center"}
        show={true}
      >
        <Modal.Header className={"my-modal"}>
          <Modal.Title style={{ fontFamily: "Bubblegum Sans" }}>
            404_
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>Uh oh... Vi kan ikke finne det du leter etter.</Modal.Body>
        <Modal.Footer className={"d-flex justify-content-center my-modal"}>
          <CapsuleButtonYellow
            className={"justify-content-center align-content-center"}
            buttonText={"Tilbake til startsiden"}
            onClick={() => navigate("/")}
          />
        </Modal.Footer>
      </Modal>
    </div>
  );
}
