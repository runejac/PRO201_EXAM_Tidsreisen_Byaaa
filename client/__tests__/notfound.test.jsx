/**
 * @jest-environment jsdom
 */

import React from "react";
import { NotFound } from "../src/components/NotFound.jsx";
import { fireEvent, render } from "@testing-library/react";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useNavigate: () => mockedUsedNavigate,
}));

describe("NotFound", () => {
  it("shows snapshot", async () => {
    const component = render(<NotFound />);
    expect(component).toMatchSnapshot();
  });

  it("simulates click", async () => {
    const { getByText } = render(<NotFound />);
    fireEvent.click(
      getByText(/Tilbake til startsiden/i, { selector: "button" })
    );
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    expect(mockedUsedNavigate).toHaveBeenCalledWith("/");
  });
});
