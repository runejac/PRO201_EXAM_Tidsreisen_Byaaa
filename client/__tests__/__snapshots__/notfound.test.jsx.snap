// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound shows snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div>
      <div
        id="background-modal-notfound"
      />
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade d-flex justify-content-center align-items-center modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="my-modal modal-header"
          >
            <div
              class="modal-title h4"
              style="font-family: Bubblegum Sans;"
            >
              404_
            </div>
          </div>
          <div
            class="modal-body"
          >
            Uh oh... Vi kan ikke finne det du leter etter.
          </div>
          <div
            class="d-flex justify-content-center my-modal modal-footer"
          >
            <button
              id="button-capsules-yellow"
            >
              Tilbake til startsiden
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="background-modal-notfound"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
